modin/experimental/backends/omnisci/query_compiler.py

```diff
@@ -619,17 +620,35 @@ def has_multiindex(self, axis=0):
         assert axis == 1
         return isinstance(self.columns, pandas.MultiIndex)
 
-    def get_index_name(self):
-        return self._modin_frame.get_index_name()
+    def get_index_name(self, axis=0):
+        return self.columns.name if axis else self._modin_frame.get_index_name()
 
-    def set_index_name(self, name):
-        self._modin_frame = self._modin_frame.set_index_name(name)
-
-    def get_index_names(self):
-        return self._modin_frame.get_index_names()
-
-    def set_index_names(self, names):
-        self._modin_frame = self._modin_frame.set_index_names(names)
+    def set_index_name(
+        self, name=no_default, axis=0, index=no_default, columns=no_default
+    ):
+        if index is no_default and columns is no_default:
+            if axis == 0:
+                index = name
+            else:
+                columns = name
+        if index is not no_default:
+            self._modin_frame = self._modin_frame.set_index_name(index)
+        elif columns is not no_default:
+            self.columns.name = columns
+
+    def get_index_names(self, axis=0):
+        return self.columns.names if axis else self._modin_frame.get_index_names()
```

Comment:
So this is the **main** place help us to avoid unnecessary index materialization?

Reply:
well, all changes in this commit help us to avoid unnecessary index materialization, we just don't want to access the index object when it is not necessary